actions/mongodb_explain_action.go

```diff
@@ -0,0 +1,93 @@
+// pmm-agent
+// Copyright 2019 Percona LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package actions
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/percona/percona-toolkit/src/go/mongolib/proto"
+	"github.com/percona/pmm/api/agentpb"
+	"github.com/pkg/errors"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+type mongodbExplainAction struct {
+	id     string
+	params *agentpb.StartActionRequest_MongoDBExplainParams
+}
+
+// NewMongoDBExplain creates a MongoDB  EXPLAIN query Action.
+func NewMongoDBExplain(id string, params *agentpb.StartActionRequest_MongoDBExplainParams) Action {
+	return &mongodbExplainAction{
+		id:     id,
+		params: params,
+	}
+}
+
+// ID returns an Action ID.
+func (a *mongodbExplainAction) ID() string {
+	return a.id
+}
+
+// Type returns an Action type.
+func (a *mongodbExplainAction) Type() string {
+	return "mongodb-explain"
+}
+
+// Run runs an Action and returns output and error.
+func (a *mongodbExplainAction) Run(ctx context.Context) ([]byte, error) {
+	dsn := a.params.Dsn
+	client, err := mongo.Connect(ctx, options.Client().ApplyURI(dsn))
+	if err != nil {
+		return nil, err
+	}
+
+	// Check the connection
+	err = client.Ping(ctx, nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "cannot connect to the database. Ping failed")
+	}
+	defer client.Disconnect(ctx) //nolint:errcheck
+
+	var eq proto.ExampleQuery
```

Comment:
Do we pull that dependency only for https://github.com/percona/percona-toolkit/blob/8abdc070501191bd4688a9e5baf28ccac97020e8/src/go/mongolib/proto/system.profile.go#L103-L111 ?

If yes, we probably could just add it to this file, and avoid dependency.

Reply:
Found https://github.com/percona/percona-toolkit/blob/8abdc070501191bd4688a9e5baf28ccac97020e8/src/go/mongolib/proto/system.profile.go#L121-L286 – is there a newer and simple way with the new driver?

Reply:
/cc @BupycHuk

Reply:
As far as I see we use percona-toolkit in mongo profiler too

Reply:
k, so be it